Staging/FswHfsPlus/fsw_hfsplus.c

```diff
@@ -471,6 +474,24 @@ fsw_hfsplus_vol_mount(struct fsw_hfsplus_volume *v)
     if (status)
         return status;
 
+    // Get volume label from kHFSRootFolderID thread record
+    // Not-readed volume label is not fatal error, we can proceed without it
+    status = fsw_alloc(v->catf->bt_ndsz, &btnode);
+    if (status) {
+        goto done;
+    }
+    status = fsw_hfsplus_dnid2thread(v, kHFSRootFolderID, btnode, &thread);
+
+    if (!status) {
+        label.len = fsw_u16_be_swap(thread->nodeName.length);
+        label.size = sizeof(fsw_u16) * label.len;
+        label.data = thread->nodeName.unicode;
+        label.type = FSW_STRING_TYPE_UTF16_BE;
+        fsw_strdup_coerce(&v->g.label, FSW_STRING_TYPE_UTF16, &label);
+    }
+
+    fsw_free(btnode);
+done:
     return FSW_SUCCESS;
```

Comment:
Hrm, is it intended to return success always? Makes little sense to me.

Reply:
Upward this `return FSW_SUCCESS` there're several  `if (status) return status;` for fatal cases when the volume wasn't mounted. 
But if the volume was mounted, all operations were successful (reading volume header, creating root, etc.) except label reading finished with an error, we're continuing with FSW_SUCCESS. 
Maybe it's a wrong concept and we should return error on volume label reading fails (when non-readed label is fatal failure)

Reply:
Well, if the label is not read, what will the caller get? Empty or garbage?

Reply:
The caller will get an empty string, as it was initialized from `fsw_mount()`, fsw_core.c

Just noticed that when `fsw_strdup_coerce()` fails the label type can be set not  to empty (utf16), my mistake. I'll fix it

Reply:
Changes applied, now returning empty string on label reading failure